update dependencies

derekmorr opened the pull request:
This updates scalatest, joda, scalacheck, and scalacheck-shapless to pick up bug fixes.

derekmorr committed:
update dependencies

ruippeixotog commented:
There seems to be more dependencies with outdated versions:

```
Found 5 dependency updates for pureconfig
  com.lihaoyi:ammonite:test    : 0.8.2  -> 0.8.5
  org.joda:joda-convert:test   : 1.8    -> 1.8.1
  org.scala-lang:scala-library : 2.12.1 -> 2.12.2
  org.scalatest:scalatest:test : 3.0.1  -> 3.0.3
  org.tpolecat:tut-core:test   : 0.4.8            -> 0.5.2
```

Can you also update those? Additionally, Scala 2.11.8 can be updated to Scala 2.11.11.

derekmorr committed:
update ammonite-repl to 0.8.5

derekmorr committed:
update sbt; scala; scalatest; jodaconvert

derekmorr commented:
I've updated everything except tut. Tut 0.5 changes to an sbt autoplugin and changed settings. I'll need to fiddle with `build.sbt` some to get the sub-projects to enable tut (the only way I was able to get it working was to add `enablePlugins(TutPlugin)` to each project. I'm sure there's a cleaner way).

derekmorr commented:
I forgot to update the Travis file. Will do that when I get home.

derekmorr committed:
update scala versions for travis

remove unused CanBuildFrom parameter

set Xlint based on scalac version; adapted from cats

update example project to 0.7 and scala 2.12

derekmorr committed:
update example project sbt and scalariform

derekmorr commented:
I updated the example project for 2.12, removed unused imports, and updated sbt and scalariform.

ruippeixotog reviewed (approved):
LGTM! Thanks for doing this.

leifwickland commented on .travis.yml:
```diff
@@ -1,8 +1,8 @@
 language: scala
 scala:
     - 2.10.5
```
Why not 2.10.6?

derekmorr replied:
I pushed a new commit that upgrades to 2.10.6.

leifwickland commented on build.sbt:
```diff
@@ -41,14 +41,28 @@ lazy val allVersionCompilerLintSwitches = Seq(
+  "-Ywarn-numeric-widen" // In 2.10 this produces a strange spurious error
+)
+
+// Scala 2.12.2 has excessive warnings about unused implicits
```
Is this the issue that you're talking about here?
https://github.com/scala/bug/issues/10270
If so, would you be willing to add a link to it, so we can tell in the future if this problem gets fixed?

derekmorr replied:
That could be it. I didn't search for a bug, I just saw lots of compiler errors when I upgraded to 2.12.2. I saw cats has similar issues; their discussion is here - https://github.com/typelevel/cats/pull/1632/files/92a1989c88fc44d6da145b6e0fee3ec24c099ad5#diff-fdc3abdfd754eeb24090dbd90aeec2ceR574

leifwickland reviewed (approved):
@derekmorr 
I've got a couple of nitpicks. Feel free to merge either way.

derekmorr committed:
add link to underlying bug for 2.12.2 Xlint

derekmorr committed:
update to scala 2.10.6; switch example to 2.12
